cmd/clusterctl/cmd/get_kubeconfig.go

```diff
@@ -59,6 +63,8 @@ func init() {
 		"Path to the kubeconfig file to use for accessing the management cluster. If unspecified, default discovery rules apply.")
 	getKubeconfigCmd.Flags().StringVar(&gk.kubeconfigContext, "kubeconfig-context", "",
 		"Context to be used within the kubeconfig file. If empty, current context will be used.")
+	getKubeconfigCmd.Flags().BoolVar(&gk.userKubeconfig, "user", true,
+		"Fo providers that generates a user specific kubeconfig, it returns the user specific kubeconfig if set to true, else returns system generated kubeconfig.")
```

Comment:
I think we should expand a little bit the semantics of this flag:
- by default it selects the kubeconfig for the users
- it allows to fetch the cluster kubeconfig when specifically required to